ethereum/test/pyth.js

```diff
@@ -298,45 +294,27 @@ contract("Pyth", function () {
         );
         let parsed = await this.pythProxy.parseBatchPriceAttestation(rawBatch);
 
-        // Check the header
-        assert.equal(parsed.header.magic, magic);
-        assert.equal(parsed.header.versionMajor, versionMajor);
-        assert.equal(parsed.header.versionMinor, versionMinor);
-        assert.equal(parsed.header.payloadId, 2);
-
-        assert.equal(parsed.nAttestations, RAW_BATCH_ATTESTATION_COUNT);
-        assert.equal(parsed.attestationSize, RAW_PRICE_ATTESTATION_SIZE);
-
-        assert.equal(parsed.attestations.length, parsed.nAttestations);
+        assert.equal(parsed.length, RAW_BATCH_ATTESTATION_COUNT);
 
-        for (var i = 0; i < parsed.attestations.length; ++i) {
-            const prodId =
-                "0x" + (i + 1).toString(16).padStart(2, "0").repeat(32);
+        for (var i = 0; i < parsed.length; ++i) {
             const priceByte = 255 - ((i + 1) % 256);
             const priceId =
                 "0x" + priceByte.toString(16).padStart(2, "0").repeat(32);
 
-            assert.equal(parsed.attestations[i].productId, prodId);
-            assert.equal(parsed.attestations[i].priceId, priceId);
-            assert.equal(parsed.attestations[i].price, priceVal);
-            assert.equal(parsed.attestations[i].conf, 101);
-            assert.equal(parsed.attestations[i].expo, -3);
-            assert.equal(parsed.attestations[i].emaPrice, -42);
-            assert.equal(parsed.attestations[i].emaConf, 42);
-            assert.equal(parsed.attestations[i].status, 1);
-            assert.equal(parsed.attestations[i].numPublishers, 123212);
-            assert.equal(parsed.attestations[i].maxNumPublishers, 321232);
-            assert.equal(
-                parsed.attestations[i].attestationTime,
-                attestationTime
-            );
-            assert.equal(parsed.attestations[i].publishTime, publishTime);
-            assert.equal(parsed.attestations[i].prevPublishTime, 0xdeadbabe);
-            assert.equal(parsed.attestations[i].prevPrice, 0xdeadfacebeef);
-            assert.equal(parsed.attestations[i].prevConf, 0xbadbadbeef);
+            assert.equal(parsed[i].priceId, priceId);
+            assert.equal(parsed[i].priceInfo.price.price, priceVal);
+            assert.equal(parsed[i].priceInfo.price.conf, 101);
+            assert.equal(parsed[i].priceInfo.price.expo, -3);
+            assert.equal(parsed[i].priceInfo.emaPrice.expo, -3);
+
+            assert.equal(parsed[i].priceInfo.emaPrice.price, -42);
+            assert.equal(parsed[i].priceInfo.emaPrice.conf, 42);
+
+            assert.equal(parsed[i].priceInfo.emaPrice.publishTime, publishTime);
+            assert.equal(parsed[i].priceInfo.price.publishTime, publishTime);
```

Comment:
can you also add a test here for the case where the status is unknown?

Reply:
There is already a test for it. It didn't need anychange as it uses end to end just like this one. 

https://github.com/pyth-network/pyth-crosschain/blob/main/ethereum/test/pyth.js#L695
I had added this when I added this logic and it's a good kind of test that needs no change when you change internals  :D